filesystems/blob/src/main/scala/cromwell/filesystems/blob/BlobPathBuilderFactory.scala

```diff
@@ -55,60 +120,43 @@ object BlobTokenGenerator {
   }
 }
 
-case class WSMBlobTokenGenerator(container: String, endpoint: String, workspaceId: String, workspaceManagerURL: String) extends BlobTokenGenerator {
-  def getAccessToken: String = {
-    throw new NotImplementedError
-  }
+case class WSMBlobTokenGenerator(container: BlobContainerName, endpoint: EndpointURL, workspaceId: WorkspaceId, workspaceManagerURL: WorkspaceManagerURL) extends BlobTokenGenerator {
+  def generateAccessToken: Try[AzureSasCredential] = Failure(new NotImplementedError)
 }
 
-case class NativeBlobTokenGenerator(container: String, endpoint: String) extends BlobTokenGenerator {
-  def getAccessToken: String = {
-    val storageAccountName = BlobPathBuilder.parseStorageAccount(BlobPathBuilder.parseURI(endpoint)) match {
-      case Some(storageAccountName) => storageAccountName
-      case _ => throw new Exception("Storage account could not be parsed from endpoint")
-    }
+case class NativeBlobTokenGenerator(container: BlobContainerName, endpoint: EndpointURL) extends BlobTokenGenerator {
 
-    val profile = new AzureProfile(AzureEnvironment.AZURE)
-    val azureCredential = new DefaultAzureCredentialBuilder()
-      .authorityHost(profile.getEnvironment.getActiveDirectoryEndpoint)
+  private val azureProfile = new AzureProfile(AzureEnvironment.AZURE)
+  private def azureCredentialBuilder = new DefaultAzureCredentialBuilder()
+      .authorityHost(azureProfile.getEnvironment.getActiveDirectoryEndpoint)
       .build
-    val azure = AzureResourceManager.authenticate(azureCredential, profile).withDefaultSubscription
-
-    val storageAccounts = azure.storageAccounts()
-    val storageAccount = storageAccounts
-      .list()
-      .asScala
-      .find(_.name == storageAccountName)
-
-    val storageAccountKeys = storageAccount match {
-      case Some(value) => value.getKeys.asScala.map(_.value())
-      case _ => throw new Exception("Storage Account not found")
-    }
-
-    val storageAccountKey = storageAccountKeys.headOption match {
-      case Some(value) => value
-      case _ => throw new Exception("Storage Account has no keys")
-    }
-
-    val keyCredential = new StorageSharedKeyCredential(
-      storageAccountName,
-      storageAccountKey
-    )
-    val blobContainerClient = new BlobContainerClientBuilder()
-      .credential(keyCredential)
-      .endpoint(endpoint)
-      .containerName(container)
-      .buildClient()
-
-    val blobContainerSasPermission = new BlobContainerSasPermission()
-      .setReadPermission(true)
-      .setCreatePermission(true)
-      .setListPermission(true)
-    val blobServiceSasSignatureValues = new BlobServiceSasSignatureValues(
-      OffsetDateTime.now.plusDays(1),
-      blobContainerSasPermission
-    )
-
-    blobContainerClient.generateSas(blobServiceSasSignatureValues)
+  private def azure = AzureResourceManager.authenticate(azureCredentialBuilder, azureProfile).withSubscription("62b22893-6bc1-46d9-8a90-806bb3cce3c9")
```

Comment:
Let's not check in this subscription number. :) Does it need to be included in the config?

Reply:
A config is a good idea! I have made this mistake a few times and the default doesn't seem to be working for me.